pages/api/uploadAttachments.ts

```diff
@@ -0,0 +1,56 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+import { promises as fs } from 'fs'
+import path from 'path'
+import formidable, { File } from 'formidable'
+
+export const config = {
+  api: {
+    bodyParser: false
+  }
+}
+
+type ProcessedFiles = Array<[string, File]>;
+
+const uploadAttachments = async (req: NextApiRequest, res: NextApiResponse) => {
+  let status = 200
+  let resultBody = { status: 'ok', message: 'Files were uploaded successfully' }
+
+  // Get files using formidable
+  const files = await new Promise<ProcessedFiles | undefined>((resolve, reject) => {
+    const form = new formidable.IncomingForm()
+    const files: ProcessedFiles = []
+    form.on('file', function (field, file) {
+      files.push([field, file])
+    })
+    form.on('end', () => resolve(files))
+    form.on('error', err => reject(err))
+    form.parse(req, () => {
+      //
+    })
+  }).catch(e => {
+    console.log(e)
+    status = 500
+    resultBody = {
+      status: 'fail', message: 'Upload error'
```

Comment:
Would it maybe be better to surface the error here instead of generalizing to an upload error?

Reply:
+1 should be fine since it's an internal tool